src/proto/codec.rs

```diff
@@ -57,7 +60,7 @@ impl WebSocketProtocol {
 macro_rules! ensure_buffer_has_space {
     ($buf:expr, $space:expr) => {
         if $buf.len() < $space {
```

Comment:
Should this not be `$buf.capacity()`?

Reply:
No, this is purely about the data that is currently available to ensure that we can index further in the payload.